### PR TITLE
Add CovarianceMixin for multicomponent models classes

### PR DESCRIPTION
**Pull request by QRemy**
Move the covariance related properties to a CovarianceMixin class and use it for  all multi-component models classes including CompoundSpectralModel.

### CODE REVIEW

**Review by adonath — COMMENTED** [review]
Thanks @QRemy, this looks reasonable to me! However I think we should only back port the bugfix and not the new mixin class to avoid future merge conflicts when back porting other fixes...

**Comment by QRemy** [comment]
Requires #5323 before merging.

File: gammapy/modeling/covariance.py [hunk 1]
```diff
@@ -210,3 +210,32 @@ def __str__(self):
...
+        """Covariance as a `~gammapy.modeling.Covariance` object."""
+        self._check_covariance()
+
+        for model in self._models:
+            self._covariance.set_subcovariance(model.covariance)
+
+        return self._covariance
+
+    @covariance.setter
+    def covariance(self, covariance):
```
Comment (bkhelifi) [inline gammapy/modeling/covariance.py:235]: Can you add a docstring defining the type of `covariance`

**Review by bkhelifi — CHANGES_REQUESTED** [review]
Thanks for this code cleaning and rationalization. Beyond minor comments on docstrings, it looks ok for me.

File: gammapy/modeling/covariance.py (hunk 1 above)
Comment (QRemy) [reply gammapy/modeling/covariance.py:244]: The setter just display the same output as its associated property and the type of `Covariance` is described there. I think we have the same behaviour everywhere. Probably better not change too much docstring in a bugfix that has to beck backported.

**Pull request merged**